app/src/main/java/com/example/habitsmasher/HabitList.java

```diff
@@ -144,7 +145,12 @@ public void deleteHabit(Context context,
                             Habit habitToDelete,
                             int habitPosition) {
         // delete locally
-        deleteHabitLocal(habitPosition);
+        try {
+            deleteHabitLocal(habitPosition);
+        } catch (IndexOutOfBoundsException e) {
+            Log.d("HabitList", "Delete failed, index out of bounds.", e);
+        }
```

Comment:
Adding this in the case where the user swipes too fast before the local list can be updated (so the indexes don't match up). So hopefully, the app doesn't crash, but instead, we get a nice log message to debug.